spinnaker/spinnaker.go

```diff
@@ -479,12 +479,35 @@ func (s Spinnaker) asgs(appName, account, clusterName string) (result []spinnake
 	return asgs, nil
 }
 
-// CloudProvider returns the cloud provider for a given account
-func (s Spinnaker) CloudProvider(account string) (provider string, err error) {
-	url := s.accountURL(account)
+// CloudProvider returns the cloud provider for a given account name
+func (s Spinnaker) CloudProvider(name string) (provider string, err error) {
+	account, err := s.account(name)
+	if err != nil {
+		return "", err
+	}
+
+	if account.CloudProvider == "" {
+		return "", errors.New("no cloudProvider field in response body")
+	}
+
+	return account.CloudProvider, nil
+}
+
+// account represents a spinnaker account
+type account struct {
+	CloudProvider string `json:"cloudProvider"`
+	Name          string `json:"name"`
+	Error         string `json:"error"`
+}
+
+// account returns an account by its name
+func (s Spinnaker) account(name string) (*account, error) {
+	url := s.accountsURL(true)
 	resp, err := s.client.Get(url)
+
+	// Usual HTTP checks
 	if err != nil {
-		return "", errors.Wrap(err, fmt.Sprintf("http get failed at %s", url))
+		return nil, errors.Wrap(err, fmt.Sprintf("http get failed at %s", url))
```

Comment:
suggest to use `errors.Wrapf()` and not use fmt.Sprintf for error message creating
https://godoc.org/github.com/pkg/errors#Wrapf